topic05-git-advanced/book/01.01.md

```diff
@@ -0,0 +1,105 @@
+# (Recap) Saving Changes and Branching
+
+## Saving
+
+Saving changes is a two-step process in Git:
+
+```bash
+# Stages changes to the staging area
+git add
+
+# Commits a snapshot of changes to the local repo
+git commit
+```
+
+## Adding
+
+* Adds changes to the staging area
+* Does not really affect the working tree
+* Changes are not recorded until committed
+
+```bash
+# Add just these file(s)
+git add app.js
+
+# Add all changed files in this directory and sub-dirs
+git add . 
+
+# Glob and stage a pattern of files
+git add **/*.js
+
+# Unstage the changes to app.js
+git reset app.js
+```
+
+## Commiting
+
+* Commits the staged snapshot to the local repository
+* Include a meaningful commit message
+* Logically group your changes into separate commits
+
+```bash
+# Commits the staged snapshot - will open editor for a message
+git commit
+
+# Commits with a message
+git commit -m "My change"
```

Comment:
On line 38 we say "Include a meaningful commit message". I don't think there's ever a case where this would be that. :)

I always try to stick to the common convention of having a verb in the imperative as the first word in the summary line of a commit (and having the summary line max 50 chars). E.g. "Make a meaningful change" or "Fix the file extension" or whatever